### ISSUE TITLE
IllegalArgumentException when testing classical MDS's method getCoodinates()

### ISSUE THREAD

**Issue by weiruchenai** [haifengl/smile#421]
## Expected behaviour

test classical MDS's method getCoodinates(), inpult is  double[][] array = {{0,0,9,0,1},{0.9,0,0.5},{0,1,0,5,0}} ，the expect output is the principal coordinates  of data.

## Actual behaviour

The algorithm throws an exception when creating object MDS. Here is the stack trace,
>Exception in thread "main" java.lang.IllegalArgumentException: Some of the first 2 eigenvalues are < 0.

## Code snippet

```
public class MdsTest {
    public static void main(String[] args) {
        double[][] array = {{0,0.9,0.1},{0.9,0,0.5},{0.1,0.5,0}};
        MDS mds = new MDS(array,2,false);
        System.out.println(Arrays.deepToString(mds.getCoordinates()));
    }
}
```
## Input data

double[][] array = {{0,0.9,0.1},{0.9,0,0.5},{0.1,0.5,0}}

## Information

- JAVA SDK11
- 1.5.2
- Window(ItelliJ IDEA)

**Comment by haifengl** [comment]
Why don't use a real data set? I don't see the meaning of this artificial data.

**Comment by weiruchenai** [comment]
Thank you for your attention ,sorry for fogetting to inform that the error has been solved. Well, I'm using MDS to do dimensionality reduction of high-dimensional document vectors, I've got disimilarities between each other, but when creating the MDS object I found that if  the input  matrix(double[][]) is more than 900-dimension, an error was throwed 
```
    MDS mds = new MDS(array,2,true);
```
```
java.lang.IllegalStateException: Too many iterations in hqr
    at smile.math.matrix.JMatrix.hqr(JMatrix.java:1989)
    at smile.math.matrix.JMatrix.eig(JMatrix.java:1326)
    at smile.mds.MDS.<init>(MDS.java:172)
```
I can't really understand the sorce code. Does the input have a limitation or someting?Is there any solution? Once again, Thank you for your attention!
